source/Irrlicht/FW_Mutex.h

```diff
@@ -198,7 +224,53 @@ not using add because we atomically wait for value to be 0 before swapping
 **/
 #define FW_AtomicCounterMagicBlockVal 0x1000000
 
-#if _MSC_VER && !__INTEL_COMPILER
+#if defined(FW_MUTEX_H_CXX11_IMPL)
+
+#define FW_AtomicCounter std::atomic_int
+inline void FW_AtomicCounterIncr(FW_AtomicCounter &lock)
+{
+	if (lock.fetch_add(1, std::memory_order_release)+1 > FW_AtomicCounterMagicBlockVal)
+		while (lock >= FW_AtomicCounterMagicBlockVal) ;
+}
+inline void FW_AtomicCounterDecr(FW_AtomicCounter &lock)
+{
+	lock.fetch_sub(1, std::memory_order_acquire);
+}
+inline void FW_AtomicCounterBlock(FW_AtomicCounter &lock)
+{
+	int zero = 0;
+	while (!lock.compare_exchange_weak(zero, FW_AtomicCounterMagicBlockVal, std::memory_order_acquire))
```

Comment:
more than two threads can use this atomic lock so please use std::memory_order_acq_rel